src/test/java/org/folio/circulation/domain/OverdueFineCalculatorServiceTest.java

```diff
@@ -0,0 +1,344 @@
+          {false, 1.0, interval, 10.0, 1.0, interval, 10.0, 5 * minutesInInterval, 5.0});
+        parameters.add(new Object[]
+          {false, 1.0, interval, 10.0, 1.0, interval, 10.0, 15 * minutesInInterval, 10.0});
+        parameters.add(new Object[]
+          {true, 1.0, interval, 10.0, 1.0, interval, 10.0, 5 * minutesInInterval, 5.0});
+        parameters.add(new Object[]
+          {true, 1.0, interval, 10.0, 1.0, interval, 10.0, 15 * minutesInInterval, 10.0});
+      });
+    return parameters;
+  }
+
+  @Before
+  public void setUp() {
+    overdueFinePolicyRepository = mock(OverdueFinePolicyRepository.class);
+    accountRepository = mock(AccountRepository.class);
+    itemRepository = mock(ItemRepository.class);
+    feeFineOwnerRepository = mock(FeeFineOwnerRepository.class);
+    feeFineRepository = mock(FeeFineRepository.class);
+    overduePeriodCalculatorService = mock(OverduePeriodCalculatorService.class);
+
+    overdueFineCalculatorService = new OverdueFineCalculatorService(
+      overdueFinePolicyRepository, accountRepository, itemRepository,
+      feeFineOwnerRepository, feeFineRepository, overduePeriodCalculatorService);
+  }
+
+  @Test
+  public void shouldNotCreateFeeFineRecordWhenLoanIsNotProvided() {
+    CheckInProcessRecords records = mock(CheckInProcessRecords.class);
+    when(records.getLoan()).thenReturn(null);
+
+    overdueFineCalculatorService.calculateOverdueFine(records);
+
+    verifyNoInteractions(accountRepository);
+  }
+
+  @Test
+  public void shouldCreateFeeFineRecordWhenAmountIsPositive()
+    throws ExecutionException, InterruptedException {
+    Loan loan = createLoan();
+
+    when(overdueFinePolicyRepository.findOverdueFinePolicyForLoan(any()))
+      .thenReturn(CompletableFuture.completedFuture(Result.succeeded(loan)));
+    when(overduePeriodCalculatorService.getMinutes(any(), any()))
+      .thenReturn(CompletableFuture.completedFuture(Result.succeeded(periodCalculatorResult)));
+    when(itemRepository.fetchItemRelatedRecords(any()))
+      .thenReturn(CompletableFuture.completedFuture(Result.succeeded(createItem())));
+    doReturn(CompletableFuture.completedFuture(Result.succeeded(createFeeFineOwner())))
+      .when(feeFineOwnerRepository).getFeeFineOwner(SERVICE_POINT_ID.toString());
+    doReturn(CompletableFuture.completedFuture(Result.succeeded(createFeeFine())))
+      .when(feeFineRepository).getOverdueFine(eq(FEE_FINE_OWNER_ID.toString()));
+
+    CheckInProcessRecords records = new CheckInProcessRecords(
+      CheckInByBarcodeRequest.from(createCheckInByBarcodeRequest()).value())
+      .withLoan(loan);
+
+    overdueFineCalculatorService.calculateOverdueFine(records).get();
+    verify(accountRepository, times(1)).create(any());
+
+    ArgumentCaptor<Account> argument = ArgumentCaptor.forClass(Account.class);
+    verify(accountRepository).create(argument.capture());
+    assertEquals(FEE_FINE_OWNER_ID.toString(), argument.getValue().getOwnerId());
```

Comment:
Personally I would use some whitespace to separate the capture declaration, the mock verification and the assertions

Reply:
Done